### PR TITLE
Semantic versioning for osmosis v4.x

### PR DESCRIPTION
**Pull request by faddat**
Closes: #621 

## Description

I found an import of app.go that wasn't changed to v4, and changed that.  

Also, I changed some import paths in the .proto files and then ran make proto-all.

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by faddat** [comment]
I'll fix up the linting tomorrow :)

File: proto/osmosis/gamm/v1beta1/genesis.proto
Comment (mattverse) [inline proto/osmosis/gamm/v1beta1/genesis.proto:18]: I think this part is missed for the sem ver!

**Review by mattverse — COMMENTED** [review]
Can you replace line27 of `scripts/generate-proto.sh` with
```
cp -r ./github.com/osmosis-labs/osmosis/v*/x/* x/
``` 

I noticed that proto files were being generated in a separate directory, changing the makeFile with the line above should do the trick! 

If your busy with other stuff going on, I can also this PR over!

**Comment by faddat** [comment]
Hi Matt, we spoke last night and I said this was ready to go 😝 

.... in goland

.... on my computer (but not here)

I'll push now

**Comment by faddat** [comment]
Another thing that I did is make this version:

* use tendermint 0.34.15
* tm-db v0.6.6
* cosmos/gorocksdb v1.2.0

File: .golangci.yml
Comment (mattverse) [inline .golangci.yml:13]: Oh cool! Didn't know `maligend` was replaced with govet field alignment!

Comment (faddat) [reply .golangci.yml:13]: since it is just a bunch of errchecks making linter angry, should I:

* fix errcheck
or
* disable errcheck for those paths and call it a day?

Comment (mattverse) [reply .golangci.yml:13]: Looks like they have existed before, imo It should be fine if we call it a day and fix err check in a different PR for v4.x branch

Comment (ValarDragon) [reply .golangci.yml:13]: I think we can just never fix errcheck on v4.x and call it a day

File: app/params/proto.go
Comment (ValarDragon) [inline app/params/proto.go:2]: cref https://pkg.go.dev/cmd/go#hdr-Build_constraints

I didn't know about this changing in 1.17 until now

**Comment by ValarDragon** [comment]
LGTM once lint issues are fixed! (I believe just commenting out errcheck right?)

**Comment by mattverse** [comment]
I think just a few changes should land before merge!
- ensure cosmos-sdk stays at v0.42.9 at go.mod
- delete `github.com/osmosis-labs/osmosis/v4` directory that has been newly made, these seem to have been created when generated proto without setting paths in the proto gen script file.

**Review by mattverse — CHANGES_REQUESTED** [review]
I think just a few changes should land before merge!

ensure cosmos-sdk stays at v0.42.9 at go.mod
delete github.com/osmosis-labs/osmosis/v4 directory that has been newly made, these seem to have been created when generated proto without setting paths in the proto gen script file.

**Comment by faddat** [comment]
thanks!  I thought I'd gotten that v4 folder out.  brb :)

**Comment by mattverse** [comment]
LGTM! Should be good to merge! Thanks @faddat !

**Review by mattverse — APPROVED** [review]
LGTM! Should be good to merge! Thanks @faddat !

**Pull request merged**
